### PR TITLE
add list item to play video on kodi

### PR DESCRIPTION
**Pull request by khimaros**
this allows the user to long press on a video in any list (search, history, playlist, etc) and send it directly to Kodi. this change honors the existing `Show "Play with Kodi" option` setting and only displays on streams supported by Kore.

there is opportunity to reduce duplication in this part of the code base generally.

closes: #5157

#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- add list context menu item to "Play on Kodi"

#### Fixes the following issue(s)

- #5157

#### APK testing 

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5752846/app-debug.zip)

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java
Comment (XiangRongLin) [inline app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java:181]: You could provide this as a method in `KoreUtil` so if the logic for when to show the kodi entry is centralized and not copied over 3 places.

**Review by XiangRongLin — CHANGES_REQUESTED** [review]
If you can think of a way to add an unit test (not instrumented) that would be ideal.
Only thing i can think of is passing in service id, context and preference manager through the method and mocking the latter 2 in tests.

File: app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java:74]: Call this "play_with_kody" to stay consistent

**Review by Stypox — CHANGES_REQUESTED** [review]

**Comment by khimaros** [comment]
i've resolved all inline feedback. i'm afraid i'm not able to contribute instrumentation and test harness at this time.

**Review by XiangRongLin — DISMISSED** [review]

**Comment by khimaros** [comment]
thank you, @XiangRongLin! i believe i accidentally put @Stypox in the critical path as well, so it looks like i'll just need one more review from them before merging. thank you all for this amazing project.

File: app/src/main/java/org/schabi/newpipe/util/KoreUtil.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/KoreUtil.java:21]: This method can be also used in other places, e.g. in the player, can't it?

**Review by Stypox — CHANGES_REQUESTED** [review]
Changes look good, thank you :-D

File: app/src/main/java/org/schabi/newpipe/util/KoreUtil.java
Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/util/KoreUtil.java:21]: Do you mean places like
https://github.com/TeamNewPipe/NewPipe/blob/8fe1a76ec6f88556ff8964df09040135c9c51e1e/app/src/main/java/org/schabi/newpipe/player/VideoPlayerImpl.java#L934-L937

Comment (khimaros) [reply app/src/main/java/org/schabi/newpipe/util/KoreUtil.java:21]: this seems like a reasonable place to use the new method, but i couldn't find a way to get a `StreamInfoItem` from a `playQueue`. any suggestions?

Comment (XiangRongLin) [reply app/src/main/java/org/schabi/newpipe/util/KoreUtil.java:21]: @khimaros Instead of passing in a `StreamInfoItem` you can pass in the service id, since that is the thing you actually need. Otherwise those items should have a common superclass, that would work too, but just passing the id is preferred.

Comment (khimaros) [reply app/src/main/java/org/schabi/newpipe/util/KoreUtil.java]: that works, thanks!

**Comment by khimaros** [comment]
> Changes look good, thank you :-D

are you able to approve? it seems that merge is blocked on you and i'm not able to remove you as an approver.

**Comment by khimaros** [comment]
resolved remaining feedback in latest commit. please take another look.

**Comment by khimaros** [comment]
@Stypox thank you, are you able to approve? it seems merge is blocked on your approval and i'm not able to remove you.

**Review by TobiGr — APPROVED** [review]

**Comment by khimaros** [comment]
@TobiGr -- thank you for the approval. are you able to merge this or does @Stypox need to approve first? i'm not able to merge it myself.

**Comment by khimaros** [comment]
@Stypox -- friendly ping

**Pull request merged**
